src/main/java/pacApp/pacController/PaymentController.java

```diff
@@ -0,0 +1,261 @@
+        }
+
+        SoapConvertCurrencyConnector currencyConnector = null;
+        List<String> currencyCodeList = null;
+
+        try {
+            currencyConnector = new SoapConvertCurrencyConnector();
+            currencyCodeList = currencyConnector.getCurrencyCodesResponse();
+        } catch (Exception ex) {
+            log.error(ex.getMessage());
+        }
+
+        if (currencyCodeList == null) {
+            GenericResponse response = new GenericResponse(HttpStatus.INTERNAL_SERVER_ERROR.value(), "Currency conversion unavailable");
+            return new ResponseEntity<>(response, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        currencyCodeList.add("EUR");
+
+        String inputCurrency = payInfo.getInputCurrency();
+        String outputCurrency = payInfo.getOutputCurrency();
+
+        if (!currencyCodeList.contains(inputCurrency)) {
+            GenericResponse response = new GenericResponse(HttpStatus.BAD_REQUEST.value(),"Invalid input currency " + inputCurrency);
+            return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
+        }
+
+        if (!currencyCodeList.contains(outputCurrency)) {
+            GenericResponse response = new GenericResponse(HttpStatus.BAD_REQUEST.value(),"Invalid output currency " + outputCurrency);
+            return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
+        }
+
+        if (rental.getPrice() == null) {
+            GenericResponse response = new GenericResponse(HttpStatus.BAD_REQUEST.value(),"Rental not finished");
+            return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
+        }
+
+        Float value = Float.valueOf(rental.getPrice().floatValue());
+        Float convertedValue = null;
+
+        try {
+            convertedValue = currencyConnector.convertCurrency(value, "USD", inputCurrency);
+        } catch (Exception ex) {
+            log.error(ex.getMessage());
+        }
+
+        if (convertedValue == null) {
+            GenericResponse response = new GenericResponse(HttpStatus.INTERNAL_SERVER_ERROR.value(), "Currency conversion error");
+            return new ResponseEntity<>(response, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        try {
+            convertedValue = currencyConnector.convertCurrency(convertedValue, inputCurrency, outputCurrency);
+        } catch (Exception ex) {
+            log.error(ex.getMessage());
+        }
+
+        if (convertedValue == null) {
+            GenericResponse response = new GenericResponse(HttpStatus.INTERNAL_SERVER_ERROR.value(), "Currency conversion error");
+            return new ResponseEntity<>(response, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        log.info(convertedValue.toString());
+        Double convertedPrice = Double.valueOf(convertedValue);
+        log.info(convertedPrice.toString());
+
+        payInfo.setOutputPrice(convertedPrice);
+
+        return new ResponseEntity<>(payInfo, HttpStatus.OK);
+    }
```

Comment:
I don't think this interface has to be exposed to the frontend. The frontend just wants to book cars and not calculate fees.
For example:

User logs in
User requests cars. With the jwt token you know the username --> querie the db for his stored currency and return the cars with the corresponding currency back to the user. The conversion of the currency is done via soap request to the currency converter
The user wants to book a car and selects the one with id 2
the server books the car for this user
the user returns the car --> the server knows the user id that booked the car and therefore the currency --> converts currency per SOAP call and sends result to frontend.
e.g. User books a car with id 1, in his user profile you see his username and can corrolate the currency,. With that you return the price of the rental to him. Same thing when the user is asking for cars.